### ISSUE TITLE
Redis error in development

### ISSUE THREAD

**Issue by McFlat** [Chocobozzz/PeerTube#1222]
the error:
```
BUSY Redis is busy running a script. You can only call SCRIPT KILL or SHUTDOWN NOSAVE.
```

Seeing as Redis is a single threaded in-memory store and uses lua scripts to do operations there's only so much it can do at one time. While TaranTool is written in lua and uses fibers to offload operations in order to scale better, tarantool also provides added functionality on top of what Redis can't do.

Would it be possible to add support for switching out Redis connector for TaranTool in the near future? So in the config admin can say use tarantool or use redis for storing key/value pairs?

Here is a link to the nodejs connector for tarantool:
https://github.com/tarantool/tarantool-node-1.5

Here is a link to the error and a article about the performance differences between tarantool vs. redis:
- https://stackoverflow.com/questions/11261631/why-lua-script-block-redis-server
- https://dzone.com/articles/tarantool-v-redis-in-memory-databases-face-off

**Comment by rigelk** [comment]
Comparing Redis and TaranTool head to head, TaranTool sure sounds like a great tool and it could be a potential improvement over Redis. However we have not just the need for a connector to TaranTool, but also the need for a job queue system like https://github.com/OptimalBits/bull supporting it in NodeJs. And correct me if I'm wrong, but that doesn't seem to exist (yet).

**Comment by McFlat** [comment]
I haven't used TaranTool or bull with Redis before, so I'm not really sure I can give you any good feedback on that just yet, but I had the Redis error and figured I'd report it and since I found out about TaranTool as an alternative I mentioned it, I think it would be possible to transition over considering that TaranTool also has a queue library https://github.com/tarantool/queue/ only it's all written in Lua and so we'd probably have to add lua language on the server or possibly as the nginx Lua module. I think it would require a good amount of work to do it, but if more admins run into these Redis errors in the future we could try to see about adding support for TaranTool into PeerTube. My main concern would be that if my server gets so much traffic that Redis would start giving more errors and I would lose things in the queue or it wouldn't be able to handle the load.

I've been translating a book from English to Russian all night so that's why I'm replying after so much time, I haven't even slept all night because it's been such a fun project to translate this book.

TaranTool just seems really awesome, and I've used Redis for at least 5 years if not 8 or so, so I'm surprised they still have such bugs, but I guess the way it's built with architecture being that way it's no real surprise, I've just never really looked into how it works under the hood until yesterday.
